e2e/cypress/integration/emoji/sorted_emojis_spec.js

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Filtered emojis are sorted by recency, then begins with, then contains', () => {
+    before(() => {
+        cy.apiLogin('user-1');
+        cy.visit('/');
+        cy.clearLocalStorage(/recent_emojis/);
+    });
+
+    it('should order emoji by recency', async () => {
+        // #Post a dog emoji
+        cy.postMessage(':dog:');
+
+        // #Post a cat emoji
+        cy.postMessage(':cat:');
+
+        // #Open emoji picker
+        cy.get('#emojiPickerButton').click();
+
+        // #Assert first recently used emoji has the data-test-id value of 'cat' which was the last one we sent
+        cy.get("[data-testid='emoji__item']").first().children.should('have.data-testid', 'cat');
+    });
+
+    it.only('should order recently used emoji first in alphabetical order, Followed by emoji that contain "word" in alphabetical', async () => {
+        const emojiList = [];
+
+        // #Post a guardsman emoji
+        cy.postMessage(':guardsman:');
+
+        // #Post a white small square emoji
+        cy.postMessage(':white_small_square:');
+
+        // #Open emoji picker
+        cy.get('#emojiPickerButton').click();
+
+        // #Search sma text in emoji searching input
+        cy.get("[data-test-id='emoji-input__search']").type('sma').then(() => {
```

Comment:
Please use `id` in this case.  Also, opening the emoji picker might vary depending on circumstance, please add check if it's visible before doing other steps, like:
```suggestion
        cy.get('#emojiInputSearch').should('be.visible').type('sma').then(() => {
```

Reply:
done